XMonad/Hooks/StatusBar/PP.hs

```diff
@@ -199,12 +209,12 @@ dynamicLogString pp = do
     -- window title
     wt <- maybe (pure "") (fmap show . getName) . S.peek $ winset
 
-    return $ sepBy (ppSep pp) . ppOrder pp $
+    return $! force (sepBy (ppSep pp) . ppOrder pp $
```

Comment:
Why not `return $! force $ sepBy …` ?

Reply:
Habit because there's already a `$` and I didn't feel like figuring out how they'd interact. I have a love-hate relationship with `$`.